freeipa/src/main/java/com/sequenceiq/freeipa/flow/freeipa/backup/full/CreateFullBackupHandler.java

```diff
@@ -0,0 +1,93 @@
+import com.sequenceiq.freeipa.service.stack.StackService;
+
+import reactor.bus.Event;
+
+@Component
+public class CreateFullBackupHandler extends ExceptionCatcherEventHandler<CreateFullBackupEvent> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CreateFullBackupHandler.class);
+
+    @Inject
+    private HostOrchestrator orchestrator;
+
+    @Inject
+    private GatewayConfigService gatewayConfigService;
+
+    @Inject
+    private StackService stackService;
+
+    @Inject
+    private FreeIpaNodeUtilService nodeService;
+
+    @Override
+    public String selector() {
+        return EventSelectorUtil.selector(CreateFullBackupEvent.class);
+    }
+
+    @Override
+    protected Selectable defaultFailureEvent(Long resourceId, Exception e, Event<CreateFullBackupEvent> event) {
+        LOGGER.error("Unexpected error happened during backup creation", e);
+        return new StackFailureEvent(FullBackupEvent.FULL_BACKUP_FAILED_EVENT.event(), resourceId, e);
+    }
+
+    @Override
+    protected Selectable doAccept(HandlerEvent<CreateFullBackupEvent> event) {
+        Stack stack = stackService.getByIdWithListsInTransaction(event.getData().getResourceId());
+        MDCBuilder.buildMdcContext(stack);
+        Set<Node> nodes = nodeService.mapInstancesToNodes(stack.getNotDeletedInstanceMetaDataSet());
+        OrchestratorStateParams stateParameters = createOrchestratorStateParams(stack, nodes);
+        try {
+            runBackupForNodesSequentially(nodes, stateParameters);
+            return new StackEvent(FullBackupEvent.FULL_BACKUP_SUCCESSFUL_EVENT.event(), event.getData().getResourceId());
+        } catch (CloudbreakOrchestratorFailedException | CloneNotSupportedException e) {
+            LOGGER.error("Full backup failed for node: {}", stateParameters.getTargetHostNames(), e);
+            return new StackFailureEvent(FullBackupEvent.FULL_BACKUP_FAILED_EVENT.event(), event.getData().getResourceId(), e);
+        }
+    }
+
+    private void runBackupForNodesSequentially(Set<Node> nodes, OrchestratorStateParams stateParameters)
+            throws CloudbreakOrchestratorFailedException, CloneNotSupportedException {
+        for (Node node : nodes) {
+            OrchestratorStateParams orchestratorStateParams = stateParameters.clone();
+            LOGGER.info("Run full backup for {}", node);
+            orchestratorStateParams.setTargetHostNames(Set.of(node.getHostname()));
+            orchestrator.runOrchestratorState(orchestratorStateParams);
+        }
+    }
+
+    private OrchestratorStateParams createOrchestratorStateParams(Stack stack, Set<Node> nodes) {
+        GatewayConfig primaryGatewayConfig = gatewayConfigService.getPrimaryGatewayConfig(stack);
+        OrchestratorStateParams stateParameters = new OrchestratorStateParams();
+        stateParameters.setPrimaryGatewayConfig(primaryGatewayConfig);
+        stateParameters.setState("freeipa.backup-full");
```

Comment:
in one way it would, as it would be collected in one place. From other perspective I would keep it close to the logic where it's used, as it's done in cb core too